helm/prometheus-meta-operator/templates/prometheus-rules/prometheus.rules.yml

```diff
@@ -22,3 +22,17 @@ spec:
         severity: page
         team: atlas
         topic: observability
+    ## Pages Atlas when prometheus fails to send samples to cortex
+    - alert: PrometheusFailsToCommunicateWithRemoteStorageAPI
+      annotations:
+        description: '{{`Prometheus can''t communicate with Remote Storage API at {{ $labels.url }}.`}}'
+      expr: rate(prometheus_remote_storage_samples_failed_total{app="prometheus"}[1h]) > 0.10 or rate(prometheus_remote_storage_samples_total[1h]) == 0
```

Comment:
I'm a bit concerned that this could be too noisy:

- isn't error rate of 10% quite low? I mean even if the error rate is > 50%, as long as it's succeeding eventually, that's OK, no?
- if there are no samples that means we are likely already getting paged with heartbeat or whatever as Prometheus is having issues; wouldn't this just mean now we'd be getting two alerts at the same time instead of 1?

Also, does having the range of `1h` not mean that if this fires and the cause is fixed, it'll take another hour for this to resolve?

Any chance you could test some scenarios like that?

Reply:
From what I saw, the usual error rate is below 20% so 10% was a nice number. Sadly, under 10% can mean it will never succeed as cortex only supports 24h old samples and prometheus currently does not support dropping samples. There is a design doc about this but this might not make it before prometheus 3.0.0.

There could be a lack of samples for other reasons, like prometheus taking it's time to read the wal and so on. It could be that we will get paged twice but this is still unclear so I preferred to page twice now and remove this if this is too noisy. 

Remember this will only page during working hours :D

It does not completely mean that,because the rate drops with every passing minute. From what I saw, samples are retried every hour or so so the rate of 1h is needed.

Honestly, I'm not sure I can test such a complex setup without setting a proxy to cut the connection to cortex for more than 24h and I honestly think this does not deserve this effort